src/databricks/labs/dqx/llm/README.md

```diff
@@ -0,0 +1,394 @@
+
+# Default configuration (LLM features disabled)
+config = ProfilerConfig()
+
+# Enable LLM-based PK detection via configuration
+config = ProfilerConfig(
+    llm_config=LLMConfig(
+        enable_pk_detection=True,
+        pk_detection_endpoint="databricks-meta-llama-3-1-8b-instruct"
+    )
+)
+
+runner = ProfilerRunner(ws, spark, installation, profiler, generator)
+checks, summary_stats = runner.run(InputConfig(location="catalog.schema.table"), config)
+
+# Check results
+if "llm_primary_key_detection" in summary_stats:
+    pk_info = summary_stats["llm_primary_key_detection"]
+    print(f"Detected PK: {pk_info['detected_columns']}")
+    print(f"Confidence: {pk_info['confidence']}")
+```
+
+### **Method 2: Options-Based**
+```python
+from databricks.labs.dqx.profiler.profiler import DQProfiler
+
+profiler = DQProfiler(ws)
+
+# Enable via options parameter
+summary_stats, dq_rules = profiler.profile_table(
+    "catalog.schema.table",
+    options={
+        "llm": True,  # Simple LLM enablement
+        "llm_pk_detection_endpoint": "databricks-meta-llama-3-1-8b-instruct"
+    }
+)
+
+# Or use the explicit flag
+summary_stats, dq_rules = profiler.profile_table(
+    "catalog.schema.table",
+    options={"enable_llm_pk_detection": True}
+)
+```
+
+### **Method 3: Convenience Method**
+```python
+from databricks.labs.dqx.profiler.profiler import DQProfiler
+
+profiler = DQProfiler(ws)
+
+# Use the convenience method - automatically enables LLM detection
+summary_stats, dq_rules = profiler.profile_table_with_llm_pk_detection(
+    table="catalog.schema.table",
+    llm_endpoint="databricks-meta-llama-3-1-8b-instruct",
+    sample_fraction=0.1
+)
+
+# Check if primary key was detected
+if "llm_primary_key_detection" in summary_stats:
+    pk_info = summary_stats["llm_primary_key_detection"]
+    print(f"✅ Detected PK: {pk_info['detected_columns']}")
+    print(f"Confidence: {pk_info['confidence']}")
+    
+    # Users can manually create validation rules based on detection results
+    # Example: Create a uniqueness check for the detected primary key
+    # (This would be done separately using the DQX rule creation APIs)
+```
+
+### **Method 4: Direct Detection**
+```python
+from databricks.labs.dqx.profiler.profiler import DQProfiler
+
+profiler = DQProfiler(ws)
+
+# Direct LLM-based primary key detection
+result = profiler.detect_primary_keys_with_llm(
+    table_name="customers",
+    catalog="main",
+    schema="sales",
+    llm=True,  # Explicit LLM enablement required
+    options={
+        "llm_pk_detection_endpoint": "databricks-meta-llama-3-1-8b-instruct"
+    }
+)
+
+if result and result.get("success", False):
+    print(f"✅ Detected PK: {result['primary_key_columns']}")
+    print(f"Confidence: {result['confidence']}")
+    print(f"Reasoning: {result['reasoning']}")
+else:
+    print("❌ Primary key detection failed or returned no results")
+```
+
+## ⚙️ **Configuration Options**
```

Comment:
remove the `LLMConfig Fields` section. We should only show how to configure this in config.yml and not provide implementation detail.